wallet/wallet.c

```diff
@@ -501,6 +510,17 @@ static bool deep_enough(u32 maxheight, const struct utxo *utxo,
 		if (csv_free > current_blockheight)
 			return false;
 	}
+
+	/* If the utxo is a coinbase, we over-write the maxheight
+	 * to the coinbase maxheight (current - 99) */
+	if (utxo->is_in_coinbase) {
```

Comment:
Perhaps a helper "u32 utxo_is_immature(utxo, blockheight)"?  Returns 0 if it's mature, otherwise number of blocks until it will me?